test/README.md

```diff
@@ -22,6 +22,10 @@ You can run against a specific Rails version only by passing the version name (w
 
     bundle exec rake test:env[rails51]
 
+Zsh example:
+
+    bundle exec rake test:env\[rails51\]
```

Comment:
Fantastic call out that the unescaped square brackets won't work for Zsh! One option to consider would be to just go with a single example that uses quotes and would work with all shells that we support:

```shell
bundle exec rake 'test:env[rails51]'
```

But having the separate listings is fine too.

Reply:
Didn't know this was a thing! Your option is simpler, so I will update to reflect.

Reply:
Ok cool. If you'd like to (with this PR or a separate follow-up one or we can put it on the backlog), here are all our documentation's references to `rake` that would require the quotes:

```shell
$ fd .md | xargs ug --perl-regexp 'rake .*\['                                                          
./CONTRIBUTING.md
   131:	bundle exec rake test:env[rails60]

./test/README.md
    23:	   bundle exec rake test:env[rails51]
    33:	   bundle exec rake test:env[rails60] file=test/new_relic/agent/distributed_tracing/*  # everything in this folder
    34:	   bundle exec rake test:env[rails60] file=test/new_relic/agent/tracer_state_test.rb   # single file
    35:	   bundle exec rake test:env[rails60] file=test/new_relic/agent/*_test.rb              # all *_test.rb files in this folder
    36:	   bundle exec rake test:env[rails60] file=test/new_relic/agent/distributed_tracing/*,test/new_relic/agent/datastores/*  # all files in two folders

./test/multiverse/README.md
   100:	   rake test:multiverse[agent_only]
   102:	   rake test:multiverse[rails,net_http]
   112:	rake test:multiverse[agent_only,name=test_resets_event_report_period_on_reconnect,env=0,debug]
```

Reply:
Nice thank you. Can you go over your search command? I'm only familiar with grep

Reply:
Sure.

First I used `fd`, which is a tool for finding files. It's designed to be a replacement for `find` with some modern conveniences built in. So when I wanted to find all Markdown files in the project, I ran `fd .md`.

- fd homepage: https://github.com/sharkdp/fd
- Homebrew installation: `brew install fd`

Unfortunately, my `fd` config has some sort of rule that prevented an additional file, `
./infinite_tracing/CONTRIBUTING.md`, from matching. That one should have been included in my list because it also has a single reference to a `rake` task with square brackets. I should have used `fd -u` (unlimited / no filters) to notice that.

Next I piped `fd`'s output to `xargs`. `xargs` is a built-in unix tool that constructs argument lists for the purpose of feeding them to tools that accept them. I'm using it to help prep `fd`'s output to be read by `ug`. I could have actually skipped using `xargs` because `fd` supports the `-x` (execute) switch itself, but doing so gets rid of the filenames and grouping, which I wanted.

Finally, xargs delivered the list of Markdown files one at a time to ugrep. The ugrep tool (which provides a `ug` binary) is a replacement for `grep` with support for some functionality that can make it very fast. By default, `ug` will perform a regex based search, but I turn that off for speed reasons so I explicitly passed `--perl-regexp` to turn it back on temporarily. `--perl-regexp` refers to the use of the Perl compatible regular expression syntax. Perl version 5 became the gold standard for its regex (alternatively abbreviated regexp) support. You might see "PCRE" mentioned in places, and that acronym stands for "Perl compatible regex engine". The regex pattern itself, `rake .*\[` is used to find a string that:

- contains "rake"
- followed by a space
- followed by any number of anythings
- followed by a literal opening square bracket

`ug` grepped through all of the Markdown files with the given regex pattern and spit out all the lines that match, grouped by filename.

- ugrep homepage: https://github.com/Genivia/ugrep
- Homebrew installation: `brew install ugrep`

The updated command with `fg -u` which produces an additional result is as follows:

```shell
$ fd -u .md |xargs ug --perl-regexp 'rake .*\['                                                        
./CONTRIBUTING.md
   131:	bundle exec rake test:env[rails60]

./infinite_tracing/CONTRIBUTING.md
   131:	bundle exec rake test:env[rails60]

./test/README.md
    23:	   bundle exec rake test:env[rails51]
    33:	   bundle exec rake test:env[rails60] file=test/new_relic/agent/distributed_tracing/*  # everything in this folder
    34:	   bundle exec rake test:env[rails60] file=test/new_relic/agent/tracer_state_test.rb   # single file
    35:	   bundle exec rake test:env[rails60] file=test/new_relic/agent/*_test.rb              # all *_test.rb files in this folder
    36:	   bundle exec rake test:env[rails60] file=test/new_relic/agent/distributed_tracing/*,test/new_relic/agent/datastores/*  # all files in two folders

./test/multiverse/README.md
   100:	   rake test:multiverse[agent_only]
   102:	   rake test:multiverse[rails,net_http]
   112:	rake test:multiverse[agent_only,name=test_resets_event_report_period_on_reconnect,env=0,debug]
```

Reply:
Excellent explanation. Thank you!